record_api/core.py

```diff
@@ -577,7 +623,13 @@ def __call__(self, frame, event, arg) -> Optional[Tracer]:
             return None
 
         if self.should_trace_frame(frame):
-            Stack(self, frame)()
+            stack = Stack(
+                self,
+                frame,
+                previous_stack=self.previous_stack,
+            )
+            stack()
```

Comment:
Could we have this return whether it saved anything, so we only save the stack (or the other info) if we need to?

Reply:
Sure, sounds good.